src/psyclone/tests/psyir/transformations/transformations_test.py

```diff
@@ -444,6 +444,99 @@ def create_loops(body):
     omplooptrans.validate(loops[0])
 
 
+def test_omplooptrans_apply_firstprivate(fortran_reader, fortran_writer):
+    ''' Test applying the OMPLoopTrans in cases where a firstprivate
+    clause it is needed to generate functionally equivalent code than
```

Comment:
"...generate code that is functionally equivalent to the original, serial version."

Reply:
Done